### ISSUE TITLE
virt-install with loader_secure=no produces machine with secure boot

### ISSUE THREAD

**Issue by rptb1** [virt-manager/virt-manager#576]
**Distro**: Ubuntu 22.04 LTS

Tested at commit 2ebbdb279.

The following invocations unexpectedly produce machines with secure boot enabled:

1. `virt-install --name 'secure-test' --osinfo archlinux --memory 8192 --boot uefi,loader_secure=no --disk none`
2. `virt-install --name 'secure-test' --osinfo archlinux --memory 8192 --boot firmware=efi,loader_secure=no --disk none`

The underlying cause seems to be that virsh is selecting /usr/share/ovmf/OVMF.fd and not /usr/share/OVMF/OVMF_CODE_4M.fd even though the latter is listed in `virsh domcapabilities`.  Explicitly specifying the loader does work correctly: `virt-install --name 'secure-test' --osinfo archlinux --memory 8192 --boot 'loader=/usr/share/OVMF/OVMF_CODE_4M.fd,loader.readonly=yes,loader.type=pflash,nvram.template=/usr/share/OVMF/OVMF_VARS_4M.fd,loader_secure=no' --disk none`

Inspecting the XML, it appears that virt-install is correctly specifying `<loader secure="no"/>` to Libvirt, so perhaps Libvirt is failing to make the right selection?  I'm not sure if it is the responsibility of virt-install or Libvirt itself.  If the latter, please direct me.

Related: #216 

Further package details from `dpkg-query`:
```
ii  libvirt0:amd64                             8.0.0-1ubuntu7.6 amd64        library for interfacing with different virtualization systems
ii  ovmf           2022.02-3ubuntu0.22.04.1 all          UEFI firmware for 64-bit x86 virtual machines
```

Background: I'm trying to produce a concise and portable procedure that doesn't depend on paths here https://blogs.igalia.com/berto/2022/07/05/running-the-steam-decks-os-in-a-virtual-machine-using-qemu/#comment-3803

Thanks!

**Comment by phrdina** [comment]
I installed ubuntu 22.04 LTS and tried both commands and in both cases libvirt stared the VM with `OVMF_CODE_4M.fd`.
So it seems that libvirt is doing the correct thing.

I would recommend checking the following paths to see what firmware json files you have in your system:

- `/usr/share/qemu/firmware`
- `/etc/qemu/firmware`
- `$XDG_CONFIG_HOME/qemu/firmware`

More details here:
https://gitlab.com/qemu-project/qemu/-/blob/master/docs/interop/firmware.json?ref_type=heads#L354

In my case on the freshly installed Ubuntu 22.04 LTS only the first directory contains these files:
```
-rw-r--r-- 1 root root  790 z\u00e1\u0159 13  2022 40-edk2-x86_64-secure-enrolled.json
-rw-r--r-- 1 root root  752 z\u00e1\u0159 13  2022 50-edk2-x86_64-secure.json
-rw-r--r-- 1 root root  740 z\u00e1\u0159 13  2022 60-edk2-x86_64.json
```

**Comment by rptb1** [comment]
Thanks for the response.

Yes, the machines start with OVMF_CODE_4M.fd on my system too, as can be seen from e.g.
```
rb@kiwi:~$ virsh dumpxml secure-test | grep OVMF
    <loader readonly='yes' secure='no' type='pflash'>/usr/share/OVMF/OVMF_CODE_4M.ms.fd</loader>
    <nvram template='/usr/share/OVMF/OVMF_VARS_4M.ms.fd'>/var/lib/libvirt/qemu/nvram/secure-test_VARS.fd</nvram>
```

However

![Screenshot from 2023-10-12 06-29-33](https://github.com/virt-manager/virt-manager/assets/419807/44871a1f-be18-4f83-9dc6-76726a778426)

So I assume the problem is the initialization of /var/lib/libvirt/qemu/nvram/secure-test_VARS.fd

**Comment by phrdina** [comment]
Hi, sorry for late replay. I did look at it again and you are correct, that command line will use firmware with secure boot enabled. I did miss the `ms` part in the firmware name.

The correct `--boot` option to make libvirt pick firmware without secure boot on ubuntu 22.04 is:

`--boot uefi,firmware.feature0.name=secure-boot,firmware.feature0.enabled=no`

On some OSes this might not work if they don't ship firmware without secure boot but usually they would ship firmware without keys so it still works like if the secure boot is disabled. In this case the feature name needs to be `enrolled-keys`.

More details here https://libvirt.org/formatdomain.html#operating-system-booting where both feature names are described.

Note: the `loader_secure` option on your original command line controls <loader secure='...'> attribute and from the same libvirt documentation:

> Attribute secure can be used to tell the hypervisor that the firmware is capable of Secure Boot feature. It cannot be used to enable or disable the feature itself in the firmware.

We should update virt-install man page as the wording about `loader_secure` can be misleading for users.
